btp/btp.go

```diff
@@ -316,8 +316,15 @@ func (b *BTP) addRelayMessage(bu *chain.BlockUpdate, rps []*chain.ReceiptProof)
 				rm.HeightOfDst = guessHeightOfDst
 			}
 		}
-		b.log.Debugf("addRelayMessage rms:%d bu:%d rps:%d HeightOfDst:%d", len(b.rms), bu.Height, len(rps), rm.HeightOfDst)
 
+		if rps[0].Height < b.bmcLinkStatus.Verifier.Height {
+			var err error
+			if rm.BlockProof, err = b.newBlockProof(rps[0].Height, bu.Header); err != nil {
```

Comment:
@trmaphi because the offet of MTA is fixed but the offset of Verifier is changed when it has enough conditions. Then this fix will not work. We should find a solution to sync the offet of MTA with Verifier instead.

Reply:
refer to this https://github.com/icon-project/btp/blob/1bca8617c87492d0d965b05c853655657c69c493/btp/util.go#L76

You can see that we need to get Witness at b.bmcLinkStatus.Verifier.Offset

Reply:
So, When Verifer.Offset is changed, a new bug will come here right ?

Reply:
>fix a vulnerability where BMR awaits to send the block at height > BmcStatusLink.Verifier.Height to send past receipt proof.
What does it mean ?

Reply:
> So, When Verifer.Offset is changed, a new bug will come here right ?

This PR does **NOT** relate to Verifier.Offset is changed. There's no way to fix the past ReceiptProof out of offset, on BMR. Contract owner must upgrade/redeploy BMV

> fix a vulnerability where BMR awaits to send the block at height > BmcStatusLink.Verifier.Height to send past receipt proof.

Without this PR, when one of BMR purposely skip BTP Message by not include ReceiptProof to the latest Block, 
Other BMRs have to await when the Verifier and Seq changes (new BlockUpdate verifiers), then newBLockProof be added to the buffered RelayMessages. 
https://github.com/icon-project/btp/blob/icondao/btp/btp.go#L196-L201
https://github.com/icon-project/btp/blob/icondao/btp/btp.go#L342-L354

Reply:
Oh, got it